Added Custom checkmark color

aadjemonkeyrock opened the pull request:
Extended the color options to also manipulate the 'selected' image checkmark.

renssies commented on Tatsi/Views/Assets Grid/Cells/AssetCollectionViewCell.swift:
```diff
@@ -62,7 +64,8 @@ final internal class AssetCollectionViewCell: UICollectionViewCell {
         iconView.translatesAutoresizingMaskIntoConstraints = false
-        iconView.tintColor = TatsiConfig.default.colors.link
+        // At this point the config is not yet set, so no color info available
+        // iconView.tintColor = TatsiConfig.default.colors.checkMark
```
You can remove the code and comment here

renssies commented on Tatsi/Views/Assets Grid/Cells/AssetCollectionViewCell.swift:
```diff
@@ -117,6 +120,10 @@ final internal class AssetCollectionViewCell: UICollectionViewCell {
         self.shouldUpdateImage = false
+        
+        // Set the correct checkmark color
+        self.contentView.subviews[2].subviews[0].tintColor = self.colors?.checkMark ?? TatsiConfig.default.colors.checkMark
```
Using an index on subviews to find a view is bit risky. Users could add a view to the contentView and thus break this code or even crash the app. My suggestion would be to define the iconView as a private property on `AssetCollectionViewCell` and reference to that when setting the color. 

(You will have to manually apply this suggestion higher in the file)
```suggestion
        lazy private var selectedOverlay: UIView = {
        let view = UIView()
        view.translatesAutoresizingMaskIntoConstraints = false
        view.isHidden = true
        view.backgroundColor = UIColor.white.withAlphaComponent(0.4)
        
        view.addSubview(self.iconView)
        view.bottomAnchor.constraint(equalTo: self.iconView.bottomAnchor, constant: 3).isActive = true
        view.trailingAnchor.constraint(equalTo: self.iconView.trailingAnchor, constant: 3).isActive = true
        
        return view
    }()
    
    lazy private var iconView: SelectionIconView = {
        let iconView = SelectionIconView()
        iconView.translatesAutoresizingMaskIntoConstraints = false
        return iconView
    }()
```

Also, this is a little known thing but `.tintColor` on UIView allows a nil value (yes even in Swift where it is force-unwrapped). Supply a nil value for tint color will make the UIView search for a tint color in the view hierarchy (so it's asks the parent for a color). This allows the view to participate in the tint color system. 

To keep consitency with older versions of this library I would recommend this for setting the color:
```suggestion
        self.iconView.tintColor = self.colors?.checkMark
```

aadjemonkeyrock replied:
Implemented the suggestion

renssies commented on Tatsi/Config/TatsiColors.swift:
```diff
@@ -21,6 +21,9 @@ public protocol TatsiColors {
     var secondaryLabel: UIColor { get }
+    
+    /// The color for the checkmark when selecting an image
+    var checkMark: UIColor { get }
```
It would be better to make `checkMark` an optional. See further down for an explanation :)

```suggestion
    var checkMark: UIColor? { get }
```

aadjemonkeyrock replied:
I made the checkMark option, but unfortunately it still needs to be provided in the protocol

renssies commented on Tatsi/Config/TatsiColors.swift:
```diff
@@ -56,4 +59,12 @@ public struct TatsiDefaultColors: TatsiColors {
         }
     }()
+    
+    public let checkMark: UIColor = {
```
When making `checkMark` optional, you don't have to supply this method anymore.

aadjemonkeyrock replied:
To be compliant with the protocol I still need to provide a variable and the goal to be compatible with older versions is not resolved this way.

aadjemonkeyrock commented:
I've implemented all the suggested changes, but I'm not sure if it's what we want. For people using an older version the TatsiColors protocol has changed. If they have customized colors they need to make an adjustment. Hearing your comments that is not the behaviour that we want. 

Looking at this again, it would for me already be a huge step forward if instead of the special checkmark color the checkmark uses the 'link' color from the TatsiColors. In the current implementation it always uses the 'default' supplied, but by removing the 'checkMark' color as I implemented and replace this with the 'link' color it would be a simpler solution that would not bread any compatibility, because it's not necessary to change the TatsiColors protocol.

renssies commented:
@aadjemonkeyrock Sorry again for a very late reply! I'm about to merge the pull request and make some changes to keep compatibility for users :)
